### PR TITLE
Modified script to make it more efficient and added few conditions.

### PR DESCRIPTION
**Pull request by RohanSunkarapalli**
- Added a condition that if objecttype is set to forcing it automatically generates LDASIN files and doesn't incorporate other file types.
- Added the year condition for forcing from 1979 to 2006 that it should include minutes in the file name and for the rest of the time period it doesn't generate file names with minutes.

### CODE REVIEW

File: nwm_filenames/listofnwmfilenamesretro.py [hunk 1]
```diff
@@ -1,96 +1,121 @@
...
         pass
 
+
 def check_valid_urls(file_list):
-    with Pool(multiprocessing.cpu_count()) as p:
-        valid_file_list = p.map(check_url, file_list)
+    with ThreadPoolExecutor(max_workers=10) as executor:
+        valid_file_list = list(executor.map(check_url, file_list))
+
     return [file for file in valid_file_list if file is not None]
```
Comment (jameshalgren) [inline nwm_filenames/listofnwmfilenamesretro.py]: Can we move this to a helper function?

Comment (jameshalgren) [reply nwm_filenames/listofnwmfilenamesretro.py]: Done.

File: nwm_filenames/listofnwmfilenames.py [hunk 2]
```diff
@@ -393,36 +399,40 @@ def create_file_list(
 
 def main():
 
-    start_date = "20220822"
-    end_date = "20220824"
+    start_date = "20030402"
+    end_date = "20030420"
```
Comment (jameshalgren) [inline nwm_filenames/listofnwmfilenames.py:403]: These produce invalid files, but generate a valid signature with the test... 
e.g., 
https://storage.cloud.google.com/national-water-model/nwm.20030420/medium_range_mem5/nwm.t12z.medium_range.reservoir_5.f158.conus.nc

is not a valid file.

**Review by jameshalgren — CHANGES_REQUESTED** [review]
I combined the url check into a helper function. Can you make sure that it is properly checking the file existence per the comment?

**Comment by RohanSunkarapalli** [comment]
> I combined the url check into a helper function. Can you make sure that it is properly checking the file existence per the comment?

Yes, I've verified with the updated changes and it works.

**Comment by RohanSunkarapalli** [comment]
> I combined the url check into a helper function. Can you make sure that it is properly checking the file existence per the comment?

Yes, I've verified with the updated changes and it works.

**Comment by jameshalgren** [comment]
> > I combined the url check into a helper function. Can you make sure that it is properly checking the file existence per the comment?
> 
> Yes, I've verified with the updated changes and it works.

Seems like the built-in test, for instance, is still producing files from the Google archive for which do not exist, but the validator is saying they do. 

`python listofnwmfilenames.py` returns a list including `'https://storage.cloud.google.com/national-water-model/nwm.20030420/medium_range_mem5/nwm.t12z.medium_range.reservoir_5.f158.conus.nc'`

which is not valide.

**Review by jameshalgren — APPROVED** [review]
Check URL was getting a 200 response from the "Authenticated URL", which is accurate because that is a sign-in page it is returning. 

We adjusted the test to use the public URL, and now invalid addresses properly return a 404.

**Pull request merged**